test/powershell/engine/Module/ModuleSpecification.Tests.ps1

```diff
@@ -0,0 +1,256 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Take a hashtable and return two copies,
+# one with the key/value pair, the other without
+function PermuteHashtableOnProperty
+{
+    param([hashtable[]]$Hashtable, [string]$Key, [object]$Value)
+
+    foreach ($ht in $Hashtable)
+    {
+        $ht.Clone()
+    }
```

Comment:
Just as an aside doing `@{} + $ht` is essentially equivalent to `.Clone()`

Reply:
Aha! Which do you consider nicer (for our purposes)?